01week/datatypes.js

```diff
@@ -0,0 +1,151 @@
+// created stringToNumber function that takes a string as parameter. the Number() method takes the same string parameter and returns as a number
+
+function stringToNumber(string) {
+  return Number(string)
+}
+
+stringToNumber("3247")
+
+// Write a JavaScript program that takes in different datatypes and prints out whether they are a:
+// Boolean
+// Null
+// Undefined
+// Number
+// NaN
+// String
+
+// researched chapter 0 prep for week 1
+// typeof ()
+// How do I get return typeof Null & NaN? using NaN as parameter states NaN is a number.NaN is typeof when number is expected.
+// Null is a typeof object. assigns values to variables you dont want to be default undefined.
+
+
+// The null value is technically a primitive, the way "object" or "number" are primitives. This would typically mean that the type of null should also be "null". However, this is not the case because of a peculiarity with the way JavaScript was first defined.
+
+// In the first implementation of JavaScript, values were represented in two parts - a type tag and the actual value. There were 5 type tags that could be used, and the tag for referencing an object was 0. The null value, however, was represented as the NULL pointer, which was 0x00 for most platforms. As a result of this similarity, null has the 0 type tag, which corresponds to an object.
+// NaN The type of NaN, which stands for Not a Number is, surprisingly, a number. The reason for this is, in computing, NaN is actually technically a numeric data type. However, it is a numeric data type whose value cannot be represented using actual numbers. So, the name "Not a Number", doesn't mean that the value is not numeric. It instead means that the value cannot be expressed with numbers.---bitsofco.de
+
+function dataTypePrinter(datatype) {
+  return typeof datatype
+}
+
+dataTypePrinter(Number("hello"));
+dataTypePrinter(document.all);
+dataTypePrinter(null);
+
+
+// Write a JavaScript program that adds 2 numbers together.
+// + means add. created addTwoNumbers function that takes two parameters that need to be numbers and adds them together. checked typeof by nesting dataTypePrinter function. strings and booleans can be concatenated together but are not numbers.
+
+function addTwoNumbers(num1,num2) {
+  if (dataTypePrinter(num1) === 'number' && dataTypePrinter(num2) === 'number') {
+  return num1 + num2
+} else { return "please enter numbers"
+}
+}
+
+addTwoNumbers(3,);
+addTwoNumbers(543, 3);
+
+
+// Write a JavaScript program that runs only when 2 things are true.
+
+// researched
+// -conditional statements -if/then/else/else if
+// -logical operators- &&=and ||=or
+
+
+function twoTrue() {
+let thing1 = true;
+let thing2 = "";
+  if (thing1 && thing2) {
+    return "run"
+  }
+}
+
```

Comment:
This function evaluates if your  argument is equal to the boolean true. I need the function to evaluate if both arguments are truthy, not the boolean true.